### ISSUE TITLE
Updating unique strings to null seems to result in an internal server error

### ISSUE THREAD

**Issue by marktani** [prisma/prisma#2426]
# Bug Report

**Current behavior**

```graphql
type A {
  id: ID! @unique
  b: String! @unique
  key: String! @unique
}
```

```graphql
mutation a {
  createA(data: {
    b: "abc"
    key: "abc"
  }) {
    id
  }
}

mutation b {
  updateA(
    where: { b: "abc" }
    data: {
      key: null
    }) {
    id
  }
}
```

results in

```json
"exception": "java.util.NoSuchElementException: Bad(InvalidValueForScalarType(None,String)).get",
"stack_trace": "org.scalactic.Bad.get(Or.scala:1368)\\n org.scalactic.Bad.get(Or.scala:1305)\\n com.prisma.api.mutactions.DatabaseMutactions.updateNodeSelectorValue(DatabaseMutactions.scala:219)\\n com.prisma.api.mutactions.DatabaseMutactions.$anonfun$getMutactionsForUpdate$1(DatabaseMutactions.scala:36)\\n scala.Option.map(Option.scala:146)\\n com.prisma.api.mutactions.DatabaseMutactions.getMutactionsForUpdate(DatabaseMutactions.scala:36)\\n com.prisma.api.mutations.Update.$anonfun$prepareMutactions$1(Update.scala:35)\\n scala.util.Success.$anonfun$map$1(Try.scala:251)\\n scala.util.Success.map(Try.scala:209)\\n scala.concurrent.Future.$anonfun$map$1(Future.scala:287)\\n scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:29)\\n scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:29)\\n scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)\\n scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:140)\\n java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)\\n java.util.concurrent.ForkJoinPool$WorkQueue.pollAndExecAll(ForkJoinPool.java:1021)\\n java.util.concurrent.ForkJoinPool$WorkQueue.execLocalTasks(ForkJoinPool.java:1046)\\n java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1058)\\n java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)\\n java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)",

```

**Expected behavior?**

Not sure if this should be rejected or allowed due to `unique` constraint.

I think it should be allowed.

**Comment by do4gr** [comment]
This is a bug since it should be allowed. Following the SQL standard we do not consider multiple nulls to be a violation of the unique property.

**Comment by k00k** [comment]
I should note, please feel free to ping me on Slack, happy to work through it in real-time.

**Comment by do4gr** [comment]
Thanks for the detailed reproduction @k00k! I'll try to look into it tomorrow.

**Comment by do4gr** [comment]
Sorry for the long delay. I can reproduce it now, you're correct it only happens if you try to set the where parameter to null.

**Comment by do4gr** [comment]
A proper fix for this is available in `1.12-beta` now.

**Comment by marktani** [comment]
This is now released.
